rule/modifies_param.go

```diff
@@ -78,3 +98,32 @@ func checkParam(id *ast.Ident, w *lintModifiesParamRule) {
 		})
 	}
 }
+
+func isModifyingFunction(callExpr *ast.CallExpr) bool {
+	funcName := astutils.GoFmt(callExpr.Fun)
+	_, found := modifyingFunctions[funcName]
+	return found
+}
+
+func (w *lintModifiesParamRule) checkModifyingFunction(callExpr *ast.CallExpr) {
+	funcName := astutils.GoFmt(callExpr.Fun)
+	positions, found := modifyingFunctions[funcName]
+	if !found {
+		return
+	}
+
+	for _, pos := range positions {
+		if pos >= len(callExpr.Args) {
+			continue
+		}
+
+		if id, ok := callExpr.Args[pos].(*ast.Ident); ok && w.params[id.Name] {
+			w.onFailure(lint.Failure{
+				Confidence: 0.5, // confidence is low because of shadow variables
+				Node:       callExpr,
+				Category:   lint.FailureCategoryBadPractice,
+				Failure:    fmt.Sprintf("parameter '%s' is modified by %s", id.Name, funcName),
```

Comment:
Maybe we can highlight the low confidence in the failure message:
```suggestion
				Failure:    fmt.Sprintf("parameter '%q seems modified by %q", id.Name, funcName),
```